src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticService.cs

```diff
@@ -47,7 +48,11 @@ internal partial class DiagnosticService : IDiagnosticService
             _updateSources = [];
 
             // queue to serialize events.
-            _eventQueue = new TaskQueue(listenerProvider.GetListener(FeatureAttribute.DiagnosticService), TaskScheduler.Default);
+            _eventQueue = new AsyncBatchingWorkQueue<(IDiagnosticUpdateSource source, ImmutableArray<DiagnosticsUpdatedArgs> argsCollection)>(
+                delay: TimeSpan.Zero,
```

Comment:
i'm wary about this.  we don't really do a zero timespan anywhere.

Reply:
is it possible ti imporve the IThreadingContext stuff so we can use the Dipose cancellation token for this guy?

Reply:
`IThreadingContext` is not available in this layer, but we can do exactly the same thing by just making this MEF part disposable.

Reply:
➡️ Updated